src/scene/scene_lua.inl

```diff
@@ -736,94 +761,53 @@ namespace {
     const std::vector<SceneGraphNode*>& nodes =
         global::renderEngine->scene()->allSceneGraphNodes();
 
-    // Extract the property and node name to be searched for from name
-    bool isLiteral = false;
-    std::string propertyName;
-    std::string nodeName;
-    size_t wildPos = name.find_first_of("*");
-    if (wildPos != std::string::npos) {
-        nodeName = name.substr(0, wildPos);
-        propertyName = name.substr(wildPos + 1, name.length());
-
-        // If none then malformed regular expression
-        if (propertyName.empty() && nodeName.empty()) {
-            throw ghoul::lua::LuaError(
-                std::format(
-                    "Malformed regular expression: '{}': Empty both before and after '*'",
-                    name
-                )
-            );
-        }
-
-        // Currently do not support several wildcards
-        if (name.find_first_of("*", wildPos + 1) != std::string::npos) {
-            throw ghoul::lua::LuaError(
-                std::format(
-                    "Malformed regular expression: '{}': "
-                    "Currently only one '*' is supported",
-                    name
-                )
-            );
-        }
-    }
-    // Literal or tag
-    else {
-        propertyName = name;
-        isLiteral = true;
-    }
+    auto [nodeName, propertyName, isLiteral] = parseRegex(name);
```

Comment:
Same here. These are not the names, but the identifiers. Please update the variable names to make this clear. 

Also, please rename the input parameter from `name` to `regex` or something